### PR TITLE
glTF deck.gl example

### PR DESCRIPTION
**Pull request by ibgreen**
This is an example of using the `GLTFLoader` with the scenegraph layer.

@georgios-uber 
- I can't get the IBL lighting to work, any chance you could take a quick look?

@xintongxia 
- The `ScenegraphLayer` doesn't update after the model loads, until I interact with the viewport, are we forgetting to call `layer.setNeedsUpdate()` somewhere in the `ScenegrapLayer?

@bravecow FYI
- you requested a simpler example
- this is not luma.gl but deck.gl, however the example is much cleaner in this API. 
- Naturally the luma.gl glTF API and example still need to be simplified.

Screen Shot 2019-09-19 at 7 44 02 AM

### CODE REVIEW

File: examples/gltf-deck/app.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    super(props);
+
+    this.state = {
+      iblEnvironment: null
+    };
+  }
+
+  componentDidMount() {
+    // TODO
+    // fileDrop(this._deckRef.deckCanvas, (promise, file) => {});
```
Comment (xintongxia) [inline examples/gltf-deck/app.js]: filedrop not working?

File: examples/gltf-deck/app.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    return (
+      <DeckGL
+        height="100%"
+        initialViewState={INITIAL_VIEW_STATE}
+        controller={{type: MapController, maxPitch: 85}}
+        onWebGLInitialized={this._onWebGLInitialized.bind(this)}
+        layers={[
+          new ScenegraphLayer({
+            id: 'tile-3d-layer',
+            data: [{}],
```
Comment (xintongxia) [inline examples/gltf-deck/app.js]: Maybe we can create an issue for deck? Some layers have to get around by passing a dummy data array.

File: examples/gltf-deck/environment.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const CUBE_FACE_TO_DIRECTION = {
+  [GL.TEXTURE_CUBE_MAP_POSITIVE_X]: 'right',
+  [GL.TEXTURE_CUBE_MAP_NEGATIVE_X]: 'left',
+  [GL.TEXTURE_CUBE_MAP_POSITIVE_Y]: 'top',
+  [GL.TEXTURE_CUBE_MAP_NEGATIVE_Y]: 'bottom',
+  [GL.TEXTURE_CUBE_MAP_POSITIVE_Z]: 'front',
+  [GL.TEXTURE_CUBE_MAP_NEGATIVE_Z]: 'back'
+};
+
+export default function loadIBLEnvironment(gl) {
```
Comment (xintongxia) [inline examples/gltf-deck/environment.js]: what is IBL?

File: examples/gltf-deck/app.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  _onWebGLInitialized(gl) {
+    this.setState({iblEnvironment: loadIBLEnvironment(gl)});
+  }
+
+  render() {
+    return (
+      <DeckGL
+        height="100%"
+        initialViewState={INITIAL_VIEW_STATE}
+        controller={{type: MapController, maxPitch: 85}}
```
Comment (xintongxia) [inline examples/gltf-deck/app.js]: ```maxPitch: 85``` is different from 
```
INITIAL_VIEW_STATE = {
  ...,
  maxPitch: 60
};
````

File: examples/gltf-deck/app.js (hunk 1 above)
Comment (ibgreen) [reply examples/gltf-deck/app.js]: I just didn't implement it yet, it should load a new model.

File: examples/gltf-deck/app.js (hunk 2 above)
Comment (ibgreen) [reply examples/gltf-deck/app.js]: We could propose it, it is a bit non-standard to not have an empty data array. Maybe add a bullet to our weekly meeting doc?

File: examples/gltf-deck/environment.js (hunk 3 above)
Comment (ibgreen) [reply examples/gltf-deck/environment.js]: Image Based Lighting

File: examples/gltf-deck/app.js (hunk 4 above)
Comment (ibgreen) [reply examples/gltf-deck/app.js]: Fixed, thanks

**Pull request merged**